mbed_lstools/lstools_base.py

```diff
@@ -167,36 +165,38 @@ def list_mbeds(
 
         return result
 
-    def _fs_never(self, device, pn):
+    def _fs_never(self, device, filter_function):
         """Filter device without touching the file system of the device"""
         device['target_id'] = device['target_id_usb_id']
         device['target_id_mbed_htm'] = None
         device['platform_name'] = self.plat_db.get(device['target_id'][0:4])
-        if device['platform_name'] and not pn.match(device['platform_name']):
-            return None
-        else:
+        if not filter_function or filter_function(device):
             return device
+        else:
+            return None
 
-    def _fs_before_id_check(self, device, pn):
+    def _fs_before_id_check(self, device, filter_function):
         """Filter device after touching the file system of the device.
         Said another way: Touch the file system before filtering
         """
         self._update_device_from_htm(device)
-        if device['platform_name'] and not pn.match(device['platform_name']):
-            return None
-        else:
+        if not filter_function or filter_function(device):
             return device
+        else:
+            return None
 
-    def _fs_after_id_check(self, device, pn):
+    def _fs_after_id_check(self, device, filter_function):
         """Filter device before touching the file system of the device.
         Said another way: Touch the file system after filtering
         """
-        plat_name = self.plat_db.get(device['target_id_usb_id'][0:4])
-        if plat_name and not pn.match(plat_name):
-            return None
-        else:
+        device['target_id'] = device['target_id_usb_id']
```

Comment:
could you add “host_id” here as well, idea was to start using that as unique id because when daplink is in bootloader mode we have have only that... (could be in follow up pr also)

Reply:
yeah it belongs to own PR, just wanted to mention it :)